executor/analyze.go

```diff
@@ -171,6 +171,12 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			sc := e.ctx.GetSessionVars().StmtCtx
 			globalStats, err := statsHandle.MergePartitionStats2GlobalStats(sc, infoschema.GetInfoSchema(e.ctx), globalStatsID.tableID, info.isIndex, info.idxID)
 			if err != nil {
+				errMessage := err.Error()
+				if errMessage == "[stats] build global-level stats failed due to missing partition-level stats" {
```

Comment:
Judging an error by comparing its error message is dangerous.
You'd better create a new kind of error and use `errors.Cause(err)`.
Here is an example:
```
			if _, ok := errors.Cause(err).(*tikv.PDError); !ok {
				break
			}
```